test/testenv/testenv.go

```diff
@@ -0,0 +1,82 @@
+package testenv
+
+import (
+	"io"
+	"net"
+
+	"google.golang.org/grpc"
+	mixerpb "istio.io/api/mixer/v1"
+	"istio.io/mixer/cmd/server/cmd"
+	"istio.io/mixer/cmd/shared"
+	"istio.io/mixer/pkg/handler"
+	"istio.io/mixer/template"
+)
+
+// TestEnv interface
+type TestEnv interface {
+	io.Closer
+	CreateMixerClient() (mixerpb.MixerClient, *grpc.ClientConn, error)
+}
+
+type testEnv struct {
+	addr         string
+	mixerContext *cmd.ServerContext
+	shutdown     chan struct{}
+}
+
+// Args includes the required args to initialize Mixer server.
+type Args struct {
+	ConfigStoreURL                string
+	ConfigStore2URL               string
+	ConfigDefaultNamespace        string
+	ConfigIdentityAttributeDomain string
+}
+
+// NewTestEnv creates a TestEnv instance.
+func NewTestEnv(args *Args, adapters []handler.InfoFn) (TestEnv, error) {
+	lis, err := net.Listen("tcp", "localhost:0")
```

Comment:
Should we make '"localhost:0"' a param ?

Reply:
This means choosing a open port on loop back interface. Test developer then doesn't need to scratch their heads to figure out what port they should use.

Reply:
localhost:0 chooses a free port on loop back interface. So it has some flexibility. But it doesn't hurt to make it an argument.